emperor/support_files/js/controller.js

```diff
@@ -1,37 +1,177 @@
+  this.renderer.setSize(this.width, this.height);
   this.renderer.setClearColor(this.rendererBackgroundColor);
   this.renderer.autoClear = false;
   this.renderer.sortObjects = true;
+  this.$plotSpaceId.append(this.renderer.domElement);
 
-  this.sceneViews = [
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0, 0, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0.5 * this.width, 0, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0.5 * this.width, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height)];
+  // default decomposition view uses the full window
+  this.addView();
+
+  this.buildUI();
+};
+
+EmperorController.prototype.addView = function() {
+  if (this.sceneViews.length > 4) {
+    throw Error('Cannot add another scene plot view');
+  }
+
+  var spv = new ScenePlotView3D(this.renderer, [new DecompositionView(this.dm)],
+                                this.$plotSpaceId.attr('id'), 0, 0, 0, 0);
+  this.sceneViews.push(spv);
+
+  // this will setup the appropriate sizes and widths
+  this.resize(this.width, this.height);
 };
 
 EmperorController.prototype.resize = function(width, height){
+  // update the available space we have
   this.width = width;
   this.height = height;
-  this.sceneViews[0].resize(0, 0, 0.5 * this.width, 0.5 * this.height)
-  this.sceneViews[1].resize(0.5 * this.width, 0, 0.5 * this.width, 0.5 * this.height)
-  this.sceneViews[2].resize(0, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height)
-  this.sceneViews[3].resize(0.5 * this.width, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height)
+
+  // the area we have to present the plot is smaller than the total
+  var plotWidth = this.$plotSpaceId.width();
+
+  if (this.sceneViews.length === 1) {
+    this.sceneViews[0].resize(0, 0, plotWidth, this.height);
+  }
+  else if (this.sceneViews.length === 2) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, this.height);
+  }
+  else if (this.sceneViews.length === 3) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[2].resize(0, 0.5 * this.height, plotWidth, 0.5 * this.height);
+  }
+  else if (this.sceneViews.length === 4) {
+    this.sceneViews[0].resize(0, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[1].resize(0.5 * plotWidth, 0, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[2].resize(0, 0.5 * this.height, 0.5 * plotWidth, 0.5 * this.height);
+    this.sceneViews[3].resize(0.5 * plotWidth, 0.5 * this.height, 0.5 * plotWidth, 0.5 * this.height);
+  }
```

Comment:
This looks like alotta constants.  Think we can replace 0.5 with idk, some variable named x?